### PR TITLE
added dark photon model to SimG4Core custom physics

### PR DESCRIPTION
**Pull request by sushilchauhan**
Added dark photon interaction model to SimG4Core/CustomPhysics/ package. Its similar to Bethe-Heitler model but with a dark factor.  Dark factor could be change from gen level config. pdg-id set  to 1072000 for dark photon. Presented in simulation meeting on 15th November 2016.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
@sushilchauhan , the code is OK for me but I would ask you to fix few things:

1) new model/process classes inherit from corresponding Geant4 classes but they are not produced by Geant4 Collaboration, so Geant4 disclaimers and comments should be removed. 
2) you left some lines of code commented - implementation from the base class. It is fine when you develop but for CMSSW these commented lines should be removed.
3) in headers of new classes CMSSW convention is to use
#ifndef SimG4Core_CustomPhysics_CMSDarkPairProduction_h

**Comment by sushilchauhan** [comment]
@civanch I will update the branch with your suggestion sometime soon today. Thanks.

**Comment by civanch** [comment]
@sushilchauhan , please take into account, that 9_0_X branch is in a process of creation. All new PRs are cloned to this branch. Try to commit to 8_1_X and later we will see how to add this commit to 9_0_X (likely will not be peaked up automatically). Also you may do 7_1_X PR - it is our final target. Better if you do 7_1_X from scratch on top of most recent 7_1_X IB, so would be easy to make minor modifications if would needed.

**Comment by sushilchauhan** [comment]
@civanch, I just updated the branch used for PR for 81X. Please have a look if its fine. We will do a separate PR for 7_1_X soon as suggested.

**Comment by civanch** [comment]
@sushilchauhan , please, fix this line as well:
#define CMSDarkPairProduction_h 1

**Comment by civanch** [comment]
@sushilchauhan , also class member "count" may be removed - it is nearly impossible to use and difficult make an interpretation if non-zero.

**Comment by sushilchauhan** [comment]
@civanch , updated it with two changes you suggested.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
@sushilchauhan , it seems that PR is now what we need. Fortunately, 9_0_X branch got your commits, so nothing needs to be done more. You may create 7_1_X PR.

**Comment by civanch** [comment]
+1

File: SimG4Core/CustomPhysics/src/CustomPDGParser.cc [hunk 1]
```diff
@@ -88,6 +95,10 @@ double CustomPDGParser::s_charge(int pdg)
       	      return -sign;
 	}
 
+      if(s_isDphoton(pdg)){                                                                                                                           
+          charge = 0;
```
Comment (davidlange6) [inline SimG4Core/CustomPhysics/src/CustomPDGParser.cc]: charge is already 0 from line 86, right?
@sushilchauhan @civanch

**Comment by davidlange6** [comment]
why is SimG4Core/CustomPhysics/data/decay_darkphoton.txt needed? (its empty)

File: SimG4Core/CustomPhysics/src/CustomPDGParser.cc (hunk 1 above)
Comment (sushilchauhan) [inline SimG4Core/CustomPhysics/src/CustomPDGParser.cc]: yes its already set to zero at line 86.

**Comment by sushilchauhan** [comment]
at present massless case is consider with no decay hence that file is empty.

**Comment by davidlange6** [comment]
so lets remove the duplication. Thanks

**Comment by davidlange6** [comment]
can you add a comment in the file?  (or is no file the same as an empty file? that would be better)

**Comment by sushilchauhan** [comment]
Yes that would be better but I have to check to make sure it doesn't break anything. will update it soon.

**Comment by civanch** [comment]
@sushilchauhan , you may add a link to a talk in the description.

**Comment by sushilchauhan** [comment]
@civanch @davidlange6 , updated with suggestions from David.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
@sushilchauhan , there is a technical problem in validation (not only this PR). Some time will be needed to recover the system. Because you have implemented all comments it is possible to make 7_1_X PR.

**Comment by sushilchauhan** [comment]
@civanch , Thanks Vladimir! We will submit that soon today.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**